core-rust/core-api-server/src/core_api/handlers/status_network_configuration.rs

```diff
@@ -63,84 +64,69 @@ const ALL_ENTITY_TYPES: [EntityType; 13] = [
         ),
         EntityType::AccountComponent => (
             models::address_type::Subtype::AccountComponent,
-            models::EntityType::Component,
-            extract_length(ComponentAddress::Account),
+            models::EntityType::Account,
         ),
         EntityType::EcdsaSecp256k1VirtualAccountComponent => (
             models::address_type::Subtype::EcdsaSecp256k1VirtualAccountComponent,
-            models::EntityType::Component,
-            extract_length(ComponentAddress::EcdsaSecp256k1VirtualAccount),
+            models::EntityType::Account,
         ),
         EntityType::EddsaEd25519VirtualAccountComponent => (
             models::address_type::Subtype::EddsaEd25519VirtualAccountComponent,
-            models::EntityType::Component,
-            extract_length(ComponentAddress::EddsaEd25519VirtualAccount),
+            models::EntityType::Account,
         ),
         EntityType::IdentityComponent => (
             models::address_type::Subtype::IdentityComponent,
             models::EntityType::Identity,
-            extract_length(ComponentAddress::Identity),
         ),
         EntityType::EcdsaSecp256k1VirtualIdentityComponent => (
             models::address_type::Subtype::EcdsaSecp256k1VirtualIdentityComponent,
             models::EntityType::Identity,
-            extract_length(ComponentAddress::EcdsaSecp256k1VirtualIdentity),
         ),
         EntityType::EddsaEd25519VirtualIdentityComponent => (
             models::address_type::Subtype::EddsaEd25519VirtualIdentityComponent,
             models::EntityType::Identity,
-            extract_length(ComponentAddress::EddsaEd25519VirtualIdentity),
         ),
         EntityType::EpochManager => (
             models::address_type::Subtype::EpochManager,
             models::EntityType::EpochManager,
-            extract_length(ComponentAddress::EpochManager),
         ),
         EntityType::Validator => (
             models::address_type::Subtype::Validator,
             models::EntityType::Validator,
-            extract_length(ComponentAddress::Validator),
         ),
         EntityType::Clock => (
             models::address_type::Subtype::Clock,
             models::EntityType::Clock,
-            extract_length(ComponentAddress::Clock),
         ),
         EntityType::AccessControllerComponent => (
             models::address_type::Subtype::AccessController,
             models::EntityType::AccessController,
-            extract_length(ComponentAddress::AccessController),
         ),
     };
     models::AddressType {
         hrp_prefix: hrp_set.get_entity_hrp(&entity_type).to_string(),
-        subtype,
         entity_type: api_entity_type,
+        subtype,
         address_byte_prefix: entity_type.id().into(),
-        address_byte_length: address_length
-            .try_into()
-            .expect("address was longer than expected"),
+        address_byte_length: ADDRESS_LENGTH.try_into().unwrap(),
```

Comment:
I see that `ADDRESS_LENGTH` is 27, but an example `extract_length(PackageAddress::Normal)` (i.e. the way it was done before this PR) returns 26 for me. 

Is this intentional, or did you mean `ADDRESS_HASH_LENGTH`?

Reply:
It was previously incorrect - the full address length includes the prefix byte, so is 27 bytes.

This was due to either a previous confusion of mine around address serialization, or a change in upstream behaviour of how they serialize addresses.